cmake/BLTSetupTargets.cmake

```diff
@@ -16,30 +16,27 @@ if (NOT BLT_LOADED)
 endif()
 # If the generated TPL config file exists, include it here.
 if (EXISTS "${CMAKE_CURRENT_LIST_DIR}/BLTThirdPartyConfigFlags.cmake")
+  # Case: Imported BLT project (ie. an installed TPL loading its BLT targets)
   include("${CMAKE_CURRENT_LIST_DIR}/BLTThirdPartyConfigFlags.cmake")
 else()
+  # Case: Main BLT project (ie. a project loading it's own BLT)
   # Otherwise, configure the TPL flags.  We have to prefix these variables with
   # BLT so that they never conflict with user-level CMake variables in downstream
   # projects.
   # 
   # We have to do some checks before we overwrite these internal variables to assure
   # that we don't turn off a third-party library enabled by the dependency.
-  if (NOT DEFINED BLT_ENABLE_HIP OR NOT ${BLT_ENABLE_HIP})
-    set(BLT_ENABLE_HIP              ${ENABLE_HIP})
-  endif()
-  if (NOT DEFINED BLT_ENABLE_CUDA OR NOT ${BLT_ENABLE_CUDA})
-    set(BLT_ENABLE_CUDA             ${ENABLE_CUDA})
-  endif()
-  if (NOT DEFINED BLT_ENABLE_MPI OR NOT ${BLT_ENABLE_MPI})
-    set(BLT_ENABLE_MPI              ${ENABLE_MPI})
-  endif()
-  if (NOT DEFINED BLT_ENABLE_OPENMP OR NOT ${BLT_ENABLE_OPENMP})
-    set(BLT_ENABLE_OPENMP           ${ENABLE_OPENMP})
-  endif()
-  if (NOT DEFINED BLT_ENABLE_FIND_MPI OR NOT ${BLT_ENABLE_FIND_MPI})
-    set(BLT_ENABLE_FIND_MPI         ${ENABLE_FIND_MPI})
+  if (DEFINED BLT_ENABLE_HIP)
+    message(FATAL_ERROR "This variable should never exist at this point. Something has gone horribly wrong.")
```

Comment:
Why only a warning for this variable? I'm guessing this is for debugging?

Reply:
This was to guard against a weird state of the TPL variables existing when they shouldn't. Basically when two BLT's are loaded in the same project. But we have BLT_LOADED for this reason. It should be safe to delete but I could also just put all the variables there.

Reply:
I agree w/ @adayton1 -- it should be ~all~ [a representative sample] or nothing